src/lib/templateHelpers.ts

```diff
@@ -131,3 +132,14 @@ export const isCommonQueryParameter = (property: string): boolean =>
   property
     ? commonParameterPositions.queryParameters.includes(property.toString())
     : false;
+
+/**
+ * Checks whether a trait name is allowed to be used in the API.
+ * Currently only disallows "offset-paginated", as "OffsetPaginated" is also
+ * available, and is preferred.
+ *
+ * @param trait - Trait to check
+ * @returns true unless the trait's name is "offset-paginated"
+ */
+export const isAllowedTrait = (trait: amf.model.domain.Trait): boolean =>
```

Comment:
Why are we hardcoding it instead of checking whether a trait is UpperCamelCase or not?

Reply:
According to the API standards, traits are supposed to be camel case. Currently, `offset-paginated` is the only trait that is noncompliant. Theoretically, no more noncompliant traits will be added. However, if any are, we should treat that as an error and fix it, rather than silently ignore it (which could cause further issues).

Reply:
We can log the non-compliant traits and try to get them fixed without breaking SDK generation. This would mean that SDK generation is broken until we either get the trait fixed (which could take time) or add another exception in the code.